usb/device/USBSerial/USBCDC.h

```diff
@@ -0,0 +1,203 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef USBCDC_H
+#define USBCDC_H
+
+/* These headers are included for child class. */
+#include "USBDescriptor.h"
+#include "USBDevice_Types.h"
+
+#include "USBDevice.h"
+#include "LinkedList.h"
+
+class AsyncOp;
+
+class USBCDC: public USBDevice {
+public:
+
+    /*
+    * Constructor
+    *
+    * @param vendor_id Your vendor_id
+    * @param product_id Your product_id
+    * @param product_release Your preoduct_release
+    * @param connect_blocking define if the connection must be blocked if USB not plugged in
+    */
+    USBCDC(uint16_t vendor_id, uint16_t product_id, uint16_t product_release, bool connect_blocking);
+
+    /*
+    * Constructor
+    *
+    * @param phy USB phy to use
+    * @param vendor_id Your vendor_id
+    * @param product_id Your product_id
+    * @param product_release Your preoduct_release
+    * @param connect_blocking define if the connection must be blocked if USB not plugged in
+    */
+    USBCDC(USBPhy *phy, uint16_t vendor_id, uint16_t product_id, uint16_t product_release, bool connect_blocking);
+
+    /**
+     * Block until the terminal is connected
+     */
+    void wait_connected();
+
+    /*
+    * Send a buffer
+    *
+    * This function blocks until the full contents have been sent.
+    *
+    * @param buffer buffer to be sent
+    * @param size length of the buffer
+    * @returns true if successful false if interrupted due to a state change
+    */
+    bool send(uint8_t *buffer, uint32_t size);
+
+    /**
+     * Send what there is room for
+     *
+     * @param buffer data to send
+     * @param size maximum number of bytes to send
+     * @param actual a pointer to where to store the number of bytes sent
+     * @param now true to start data transmission, false to wait
+     */
+    void send_nb(uint8_t *buffer, uint32_t size, uint32_t *actual, bool now = true);
+
```

Comment:
not fully sure about the naming.. "_nb" == no buffer ? if so, then why is one of the params=buffer..Same for the receive as well at line 100..